### ISSUE TITLE
Ruby 3 support

### ISSUE THREAD

**Issue by utilum** [googleapis/google-api-ruby-client#907]
Currently can not install with Ruby 3. Following would fix:

```google-api-ruby-client$ git diff
diff --git a/google-api-client.gemspec b/google-api-client.gemspec
index 8b1af987..2a1e2393 100644
--- a/google-api-client.gemspec
+++ b/google-api-client.gemspec
@@ -21,7 +21,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.require_paths = %w[lib generated]
 
-  spec.required_ruby_version = '>= 2.4'
+  spec.required_ruby_version = ['>= 2.4', '< 3.1']
 
   spec.add_runtime_dependency 'representable', '~> 3.0'
   spec.add_runtime_dependency 'retriable', '>= 2.0', '< 4.0'
```

**Comment by dazuma** [comment]
This is already done in the code. We just need to do a release (which is currently held up by an unrelated issue.)
